common/test/run-bolt12_decode.c

```diff
@@ -195,7 +195,7 @@ int main(int argc, char *argv[])
 		size_t dlen;
 		struct json_escape *esc;
 
-		json_to_bool(json, json_get_member(json, t, "valid"), &valid);
+		assert(json_to_bool(json, json_get_member(json, t, "valid"), &valid));
```

Comment:
Mh I hope the compiler does not do this optimization. There is no way from our code to tell the compiler "This will never fail, so it is safe to remove" So, the static analysis should always keep this. Otherwise, we found a bug inside the compiler :)

I found while googling to make sure that I was not going to say stupid stuff by just assuming that I did a Master on this stuff that there is a way in GCC where it is possible to remove the assertion https://stackoverflow.com/a/76613201/10854225 but we need to specify extra information

Reply:
We assume assert() works in tests already, for better or worse.